web/app/Helpers/PageBuilder.php

```diff
@@ -0,0 +1,344 @@
+                $this->parameterTypes[$key] = 'AMODULE';
+
+                /** Add the POST key-value pair as a subarray of an AMODULE key. */
+                $this->allParameters['AMODULE'][$key] = $value;
+            }
+        } else {
+            /** We'll instantiate some empty keys that some calling modules expect. */
+            $this->allParameters['GET'] = [];
+            $this->allParameters['POST'] = [];
+
+            /**
+             * Nginx does a transform on the request for us.
+             *
+             * The user makes a request for a page like:
+             * `wikijump.test/testpage`
+             *
+             * But nginx transforms it and provides it to the server as:
+             * `wikijump.test/index.php?Wiki__WikiScreen/wiki_page/testpage`
+             *
+             * And so the query string looks like:
+             *          0           1         2
+             * `Wiki__WikiScreen/wiki_page/testpage`
+             *
+             * Here we break the request into its component parts.
+             */
+            $page_parameters = explode('/', $_SERVER['QUERY_STRING']);
+            $screen_type = $page_parameters[0];
+            $asset_type = $page_parameters[1];
+            $asset_name = $page_parameters[2];
+
+            /** And we populate the parameter bucket with the results. */
+            $this->parameterArray['template'] = $screen_type;
+            $this->parameterTypes['template'] = 'GET';
+            $this->parameterArray[$asset_type] = urldecode($asset_name);
+            $this->parameterTypes[$asset_type] = 'GET';
+            $this->allParameters['GET'][$asset_type] = urldecode($asset_name);
+
+            /**
+             * For a "real" GET request (one where the user provides a GET string),
+             * The value of it is available in `$_SERVER['REQUEST_URI'].
+             * However it also includes the reqested page name, which we don't need right now.
+             */
+            $get_string = explode('?', $_SERVER['REQUEST_URI'])[1];
+
+            /** We'll break the GET string into key-value pairs. */
+            $get_pairs = explode('&', $get_string);
+
+            /** Then break the key-value pairs into their respective items. */
+            foreach ($get_pairs as $get_pair) {
+                $item = explode('=', $get_pair);
+                $key = $item[0];
+                $value = $item[1];
+
+                /** We put the set in the same parameter store we're already building. */
+                $this->parameterArray[$key] = urldecode($value);
+                $this->parameterTypes[$key] = 'GET';
+                $this->allParameters['GET'][$key] = urldecode($value);
+            }
+
+            /** Lastly, POST values don't need as much manipulation. */
+            foreach ($_POST as $key => $value) {
+                /** Convert any CRLF values to LF */
+                $value = str_replace(["\r\n", "\r", "\n"], "\n", $value);
```

Comment:
Shouldn't this just be this, like the others?
```suggestion
                $value = str_replace(["\r\n", "\r"], "\n", $value);
```